lib/util/cache/package/decorator.ts

```diff
@@ -114,7 +114,10 @@ export function cache<T>({
 
     const result = await callback();
 
-    await packageCache.set(finalNamespace, finalKey, result, ttlMinutes);
+    // only cache if we got a valid result
+    if (result) {
```

Comment:
I think we should cache `null` but it ignore `undefined`. I don't want to recompute hashes every branch if one hash fails. So we should at least cache it for the run.

Reply:
ATM every single hash is separately cached so on the next branch only the failed ones are retried.  

Makes sense to only skip `undefined` in a commodity function.